src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden) {
+    super(health, name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Carnivore ||
+      (animal.hidden && animal instanceof Herbivore)
```

Comment:
Misleading conditional; this validation would allow any Animal that is not a `Carnivore`  nor a `Herbivore` through in this case, where an Animal can be only either a `Carnivore` or a `Herbivore` it works (which makes the `animal instanceof Herbivore` check redundant) but it would fail in a more generic scenario.

Bite should work if the `target is a Herbivore and is not hidden`; there is a better way to write this